### PR TITLE
School library: Refactor

### PR DESCRIPTION
**Pull request by joskalenda**
## OOP school library: refactor your code

- [ ] Learning objectives

- Describe the SOLID principles of OOP.
- Refactor your own Ruby code after gaining new knowledge.
- Apply Ruby best practices and language style guides in code.

### Features

1. Use YAGNI, KISS, and DRY rules, and Ruby style guides.
2. Follow SOLID principles.
3. Apply the "Single-responsibility Principle".
4. Make sure that the role of the main.rb file is only to start your code and ask the user for the initial choice of options (or to quit).
5. Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -4,9 +4,11 @@
 require './book'
 require './classroom'
 require './rental'
+require './print_script'
 
-class App
+class App < PrintScript
```
Comment (Meltrust) [inline app.rb:9]: Nice job! however, the SOLID could be improved, 

Please, kindly:

- [ ] The methods `create_person`, `create_student`, `create_book` and `create_rental` should be outside from this class.  That way, this class will be only for listing things (e.g. "list_all_rentals").  Please, kindly do so.
- [ ] All the requests for input and validation should be moved to `print_script.rb`, on your class `PrintScript` where your interface lives.  In this case, all the interface and user interactions should live in that class.
- [ ] The line `student = Student.new(age, 'classroom', name, parent_permission: false)` should be placed in the `create_people` method (students are people).
- [ ] Also, the line `teacher = Teacher.new(age, specialization, name)` should be placed in the `create_people` method as well (teachers are people).
- [ ] The `start_console` method which runs the main interface, should be out of the `App` class (the `App` class just provides lists with information in this case).    

## Explanation

Every class should have a single responsibility and methods that relate to the main functionality of that same class.  Also, the user input, the "puts", and the interface should be apart from the logic.

Please, refer to this [lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

## Hints

- All requests for input and input validation should be placed in your `print_script.rb` file.  That way, we comply with the single responsibility rule.

- Also, in this case, the `app.rb`, should just serve as a "lister" to populate the screen with data that the user requests (like `list_all_rentals`, etc).

- The `start_console` method could be better put in the `print_script` class.

That way, if you add more functionality in the future, you won't need to refactor the whole thing, thus making the app more easily scalable. 💪

Comment (Meltrust) [inline app.rb:9]: - [ ] Please, kindly include both authors on your `readme.md` file.  That way, both authors will take credit for this magnificent job!

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @joskalenda,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice first approach✔️ 

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (joskalenda) [reply app.rb:9]: Hello @Meltrust  Thanks for the review, the changes are made.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Jos Ntumba & Billions, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂


<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have no more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by joskalenda** [comment]
Thank you @rloterh

**Pull request merged**